packages/itwin/tree-widget/src/components/trees/imodel-content-tree/IModelContentTreeDefinition.ts

```diff
@@ -369,46 +362,26 @@ export class IModelContentTreeDefinition implements HierarchyDefinition {
     return defs;
   }
 
-  private async createCategoryChildrenQuery({
-    parentNodeInstanceIds: categoryIds,
-    parentNode,
-    instanceFilter,
-    viewType,
-  }: DefineInstanceNodeChildHierarchyLevelProps & { viewType: "2d" | "3d" }): Promise<HierarchyLevelDefinition> {
+  private async createCategoryChildrenQuery(props: DefineInstanceNodeChildHierarchyLevelProps & { viewType: "2d" | "3d" }): Promise<HierarchyLevelDefinition> {
+    const { parentNodeInstanceIds: categoryIds, parentNode, instanceFilter, viewType } = props;
+    const modelIds = parseIdsSelectorResult(parentNode.extendedData?.modelIds);
+
     // We only want to handle a category added as a child of `GeometricModel2d` or `GeometricModel3d`.
-    if (!parentNode.extendedData?.isCategory) {
-      return [];
-    }
```

Comment:
I think in this case the `isCategory` extended data item should be named more appropriately, e.g. `isUnderGeometricModel` or something like that.

Reply:
I figured since `modelIds` is not empty only for geometric model node children which are categories, `isCategory` is unnecessary. The issue in the screenshot was due to the broken test app lockfile which used old `hierarchies` version.